src/measures/fermions/ellesettete.cpp

```diff
@@ -101,27 +103,35 @@ namespace nissa
 		//loop over hits
 		for(int ihit=0;ihit<meas_pars.nhits;ihit++)
 		  {
-		    //fill the source
+		    //prepare the sources with the right structure spin x taste
 		    fill_source(source,glb_t,meas_pars.rnd_type);
-		    
+		    apply_stag_op(g5_id_source,conf,theory_pars.backfield[iflav],GAMMA_INT::GAMMA_5,GAMMA_INT::IDENTITY,source);
+		    apply_stag_op(id_g5_source,conf,theory_pars.backfield[iflav],GAMMA_INT::IDENTITY,GAMMA_INT::GAMMA_5,source);
+
 		    //compute std 2pts propagator G(m|n) ~ [D^-1(m|y) source(y)] source(n)*
 		    MINV(SIMPLE_PROP,iflav,source);
 		    
-		    //compute sequential propagator G(m|n) ~ [D^-1(m|y) source(y)] D^-1(y|n)
-		    MINV(SEQ_PROP,iflav,SIMPLE_PROP);
-		    
-		    //then glb reduction to compute the trace for the connected 3pts diagram and the 2pts
-		    SUMM_THE_TIME_TRACE_PRINT_AT_LAST_HIT(Tr_two_pts,SIMPLE_PROP,SIMPLE_PROP);
+		    //compute  2pts propagator with id x g5 at source and apply id x g5 at sink
+		    MINV(ID_G5_PROP,iflav,id_g5_source);
+		    apply_stag_op(ID_G5_PROP,conf,theory_pars.backfield[iflav],GAMMA_INT::IDENTITY,GAMMA_INT::GAMMA_5,ID_G5_PROP);
+
+		    //compute sequential propagator with id x g5 at source and apply id x g5 at sink
+		    MINV(SEQ_PROP,iflav,ID_G5_PROP);
+		    apply_stag_op(SEQ_PROP,conf,theory_pars.backfield[iflav],GAMMA_INT::IDENTITY,GAMMA_INT::GAMMA_5,SEQ_PROP);
+
+		    //then glb reduction to compute the trace for the connected 2pts, 3pts and 4pts diagrams
+		    SUMM_THE_TIME_TRACE_PRINT_AT_LAST_HIT(Tr_two_pts,SIMPLE_PROP,ID_G5_PROP);
 		    SUMM_THE_TIME_TRACE_PRINT_AT_LAST_HIT(Tr_three_pts,SIMPLE_PROP,SEQ_PROP);
+		    apply_stag_op(SEQ_PROP,conf,theory_pars.backfield[iflav],GAMMA_INT::IDENTITY,GAMMA_INT::GAMMA_5,SEQ_PROP); //s.t. at sink we have back id x id
 		    SUMM_THE_TIME_TRACE_PRINT_AT_LAST_HIT(Tr_four_pts,SEQ_PROP,SEQ_PROP);
```

Comment:
NO as in this way, you are going to obtain g5 x g5 in both source and sink. Unfortunately, you need to create another sequential propagator using the plain source, such that when you close the two props you get the required spin/taste structure

Reply:
plus, you will need to adjust the sink according